### PR TITLE
Normalize paths read from composer.json

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/maglnet/ComposerRequireChecker/issues/25

### CODE REVIEW

File: src/ComposerRequireChecker/Cli/CheckCommand.php [hunk 1]
```diff
@@ -73,14 +68,17 @@ protected function execute(InputInterface $input, OutputInterface $output) : int
...
-        $options = $this->getCheckOptions($input);
-
         $definedExtensionSymbols = (new LocateDefinedSymbolsFromExtensions())->__invoke(
             (new DefinedExtensionsResolver())->__invoke($composerJson, $options->getPhpCoreExtensions())
         );
 
         $usedSymbols = (new LocateUsedSymbolsFromASTRoots())->__invoke($sourcesASTs($getPackageSourceFiles($composerJson)));
 
+        if (!count($usedSymbols)) {
+            $output->writeln("There were no symbols found, please check your configuration.");
```
Comment (Ocramius) [inline src/ComposerRequireChecker/Cli/CheckCommand.php]: Throw an exception - easier to see traces when using `-vvv`. Also easier to test

File: src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php [hunk 2]
```diff
@@ -36,12 +36,17 @@ function (array $sourceDirs, $sourceDir) {
...
         return array_values(array_map(
             function (string $sourceDir) use ($packageDir) {
-                return $packageDir . '/' . ltrim($sourceDir, '/');
+                return $this->normalizePath($packageDir . '/' . ltrim($sourceDir, '/'));
             },
             $flattened
         ));
     }
 
+    private function normalizePath(string $path)
```
Comment (Ocramius) [inline src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php]: Return type

File: src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php [hunk 3]
```diff
@@ -36,12 +36,17 @@ function (array $sourceDirs, $sourceDir) {
...
+    {
+        return str_replace(array('\\', '/'), DIRECTORY_SEPARATOR, $path);
```
Comment (Ocramius) [inline src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php]: Use `\` to reference the constant by FQN

Comment (Ocramius) [inline src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php]: Use the short array syntax

Comment (Ocramius) [inline src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php]: The second argument can just be `'/'`, which is recognized as directory separator for all platforms by the PHP engine

Comment (Ocramius) [inline src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php]: Also, the tests wouldn't be platform dependent this way ^_^

**Comment by SpacePossum** [comment]
thanks for the review @Ocramius , hope I got all the corrections in :)

File: src/ComposerRequireChecker/FileLocator/LocateAllFilesByExtension.php [hunk 4]
```diff
@@ -25,7 +25,7 @@ private function filterFilesByExtension(Traversable $files, string $fileExtensio
 
         /* @var $file \SplFileInfo */
         foreach ($files as $file) {
-            if (! preg_match($extensionMatcher, $file->getBasename())) {
+            if ($file->isDir() || ! preg_match($extensionMatcher, $file->getBasename())) {
```
Comment (Ocramius) [inline src/ComposerRequireChecker/FileLocator/LocateAllFilesByExtension.php]: Should be checking for `! isFile` unless a particular decision was taken here to only exclude directories

Comment (Ocramius) [inline src/ComposerRequireChecker/FileLocator/LocateAllFilesByExtension.php]: Maybe a symlink test is needed here, or reliance on an existing abstraction

File: src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php [hunk 5]
```diff
@@ -36,12 +36,17 @@ function (array $sourceDirs, $sourceDir) {
...
-                return $packageDir . '/' . ltrim($sourceDir, '/');
+                return $this->normalizePath($packageDir . '/' . ltrim($sourceDir, '/'));
             },
             $flattened
         ));
     }
 
+    private function normalizePath(string $path) : string
+    {
+        return str_replace(['\\', '/'], '/', $path);
```
Comment (Ocramius) [inline src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php]: Replacing `/` with `/` is a no-op

File: src/ComposerRequireChecker/FileLocator/LocateAllFilesByExtension.php (hunk 4 above)
Comment (SpacePossum) [inline src/ComposerRequireChecker/FileLocator/LocateAllFilesByExtension.php]: I've added it a performance tweak, I take it out as it is not needed for this PR :+1:

Comment (Ocramius) [inline src/ComposerRequireChecker/FileLocator/LocateAllFilesByExtension.php]: Only functional changes please - performance only where performance improvement is needed.

This exposed another bug by the way: creating a directory or a symlink `foo.php` will likely need a test :-\

**Comment by SpacePossum** [comment]
I think all issues are addressed here, let me know me if more changes are needed :)

**Pull request merged**